homeassistant/components/tado/climate.py

```diff
@@ -500,49 +514,49 @@ def _async_update_zone_data(self):
         self._current_tado_swing_mode = self._tado_zone_data.current_swing_mode
 
     @callback
-    def _async_update_zone_callback(self):
+    def _async_update_zone_callback(self) -> None:
         """Load tado data and update state."""
         self._async_update_zone_data()
         self.async_write_ha_state()
 
     @callback
-    def _async_update_home_data(self):
+    def _async_update_home_data(self) -> None:
         """Load tado geofencing data into zone."""
         self._tado_geofence_data = self._tado.data["geofence"]
 
     @callback
-    def _async_update_home_callback(self):
+    def _async_update_home_callback(self) -> None:
         """Load tado data and update state."""
         self._async_update_home_data()
         self.async_write_ha_state()
 
-    def _normalize_target_temp_for_hvac_mode(self):
+    def _normalize_target_temp_for_hvac_mode(self) -> None:
+        def adjust_temp(min_temp, max_temp):
+            if max_temp is not None and self._target_temp > max_temp:
+                return max_temp
+            if min_temp is not None and self._target_temp < min_temp:
+                return min_temp
+            return self._target_temp
+
         # Set a target temperature if we don't have any
         # This can happen when we switch from Off to On
-        if self._target_temp is None:
+        if self._target_temp is None or not isinstance(self._target_temp, float):
```

Comment:
Why is the instance check required?

Reply:
It was a MyPy issue, that the operand couldn't check against a None. I think the inline function fixed this and I van try to refactor this.